drivers/hud_spm_report/app/models/hud_spm_report/generators/fy2020/base.rb

```diff
@@ -1428,30 +1465,115 @@ def hoh_destination_for(project_types, client_id, household_id)
+
+      nights_for_negation = all_nights.deep_dup
+      nights_for_negation = nights_for_negation.group_by { |night| night[:date] }
+
+      # Remove any project not in the project types we care about
+      # Ignore nights in a project that are on the date of exit
+      # never count days after move-in
+      all_nights.reject! do |night|
+        not_in_project_type = ! night[:project_type].in?(project_types)
+        on_exit_date = night[:date] == night[:last_date_in_program]
+        not_in_project_type || on_exit_date || night[:housed]
+      end
 
-      if days_in_selected_project_types.any?
+      # if 1b, remove any non-literally homeless at entry enrollments
+      # and any PH enrollment where entry, move-in, and exit is outside of report range
+      if line.in?([:m1b1, :m1b2])
+        all_nights.select! do |night|
+          # non-PH projects
+          # (night[:literally_homeless] && !night[:project_type].in?(PH)) ||
+          ! night[:project_type].in?(PH) ||
+          # PH literally homeless and with something during the range
+          (
+            night[:literally_homeless] &&
+            night[:project_type].in?(PH) &&
+            # opened during report range
+            (
+              (
+                night[:first_date_in_program].present? &&
+                night[:first_date_in_program] > @report.start_date &&
+                night[:first_date_in_program] <= @report.end_date
+              ) ||
+              # moved in during report range
+              (
+                night[:MoveInDate].present? &&
+                night[:MoveInDate] > @report.start_date &&
+                night[:MoveInDate] <= @report.end_date
+              ) ||
+              # exited during report range without moving in
+              (
+                night[:MoveInDate].blank? &&
+                night[:last_date_in_program].present? &&
+                night[:last_date_in_program] >= @report.start_date &&
+                night[:last_date_in_program] <= @report.end_date
+              )
+            )
+          )
+        end
+      end
+
+      # group by date so we can reject dates where stop-projects are present
+      all_nights = all_nights.group_by { |night| night[:date] }
+      # remove any days with a stop project type
+      all_nights.each do |date, nights|
+        if line.in?([:m1a1, :m1b1])
+          remove_homeless_if = nights_for_negation[date].any? { |night| night[:project_type].in?(TH) || (night[:project_type].in?(PH) && night[:housed]) }
+          nights.reject! { |night| night[:project_type].in?(ES_SH) } if remove_homeless_if
+          # NOTE: the spec says:
+          # a. For measures 1a.1 and 1b.1, time spent by clients housed in TH or PH projects negates overlapping time spent in ES and SH projects.
+          # data labe test kit seems to reject any homeless time
```

Comment:
I think the AAQ confirmed this behavior as correct? Update this to reflect the referenced explanation?